src/Autofac.Integration.WebApi.Owin/DependencyScopeHandler.cs

```diff
@@ -42,13 +42,15 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
         var dependencyScope = new AutofacWebApiDependencyScope(lifetimeScope);
         request.Properties[HttpPropertyKeys.DependencyScope] = dependencyScope;
 
-        try
-        {
-            return base.SendAsync(request, cancellationToken);
-        }
-        finally
-        {
-            request.Properties.Remove(HttpPropertyKeys.DependencyScope);
-        }
+        // Using .ContinueWith instead of try/finally because you can't
+        // have async/await in [SecurityCritical] or [SecuritySafeCritical]
+        // marked code.
+        return base
+            .SendAsync(request, cancellationToken)
+            .ContinueWith(task =>
+            {
+                request.Properties.Remove(HttpPropertyKeys.DependencyScope);
+                return task.Result;
+            });
```

Comment:
There's an analyser that indicates `ContinueWith` should always be used with an explicit scheduler? https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2008

Reply:
Weird, the analyzer didn't fire off and doesn't show in the build or VS. Let me add some of the stuff from the blog article they link in the doc there.